bottlerocket-settings-models/modeled-types/src/shared.rs

```diff
@@ -945,6 +945,90 @@ string_impls_for!(Lockdown, "Lockdown");
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// ApiCommand represents a valid Bottlerocket API Command. It stores the command as a vector of
+/// strings and ensures that the first argument is apiclient.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Default, Serialize, Deserialize)]
+pub struct ApiCommand(pub Vec<String>);
+
+impl TryFrom<Vec<String>> for ApiCommand {
+    type Error = error::Error;
```

Comment:
I don't think we should default to the shared types crate for types that aren't likely to see much reuse (like this one). They can live in the extension instead.

Reply:
I agree but it felt more general than BootstrapContainerMode placed here. I should move this one to settings-extensions and change BootstrapContainerMode to BootstrapMode and reuse it. It would then justify BootstrapContainerMode's presence in shared crate.

Reply:
Updated. ApiclientCommand is now moved to settings-extensions/bootstrap-commands/src/lib.rs. I would like another review of that file.